DevelopGitHub/GitHubExtension/GitHubExtension.Statistics.WebApi/Controllers/StatisticController.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.AspNet.Identity;
+
+namespace GitHubExtension.Statistics.WebApi.Controllers
+{
+    public class StatisticController : ApiController
+    {
+        private readonly IStatisticsQuery _statisticsQuery;
+
+        public StatisticController(IStatisticsQuery statisticsQuery)
+        {
+            this._statisticsQuery = statisticsQuery;
+        }
+
+        public HttpContext GetHttpContext
+        {
+            get { return HttpContext.Current;}
+        }
+
+        #region methods
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetFollowers)]
+        public async Task<int> GetUserFollower()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+
+            int follower = await _statisticsQuery.GetFollowerCount(userName, token);
+            return follower;
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetFollowing)]
+        public async Task<int> GetUserFollowing()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+
+            int following = await _statisticsQuery.GetFollowingCount(userName, token);
+            return following;
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetRepositoriesCount)]
+        public async Task<int> GetRepositoriesCount()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+
+            int repositoriesCount = await _statisticsQuery.GetRepositoriesCount(userName, token);
+            return repositoriesCount;
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetActivityMonths)]
+        public async Task<ICollection<string>> GetActivityMonths()
+        {
+            ICollection<string> activityMonths = await _statisticsQuery.GetActivityMonths();
+            return activityMonths;
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetRepositories)]
+        public async Task<ICollection<RepositoryModel>> GetRepositories()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+            ICollection<RepositoryModel> repositories = await _statisticsQuery.GetRepositories(userName, token);
+
+            return repositories;
+        }
+
+        [HttpGet]
+        [Route(StatisticsRouteConstants.GetCommitsRepositories)]
+        public async Task<ICollection<ICollection<int>>> GetCommitsRepositories()
+        {
+            string token = User.Identity.GetExternalAccessToken();
+            string userName = User.Identity.GetUserName();
+            HttpContext httpContext = GetHttpContext;
+
+            ICollection<ICollection<int>> commitsRepositories = await _statisticsQuery.GetCommitsRepositories(userName, token, httpContext);
```

Comment:
Ohh...no. Do't pass context as parameter. Use only important information for your method.